### PR TITLE
implement experiment features

### PR DESCRIPTION
**Pull request by AxelVivien25**

### CODE REVIEW

File: mlflow_elasticsearchstore/elasticsearch_store.py [hunk 1]
```diff
@@ -50,69 +53,88 @@ class ElasticsearchStore(AbstractStore):
...
+        ExperimentIndex.init(indices=self.indices)
+        RunIndex.init(indices=self.indices)
         super(ElasticsearchStore, self).__init__()
 
     def _hit_to_mlflow_experiment(self, hit: Any) -> Experiment:
-        return Experiment(experiment_id=hit.meta.id, name=hit.name,
-                          artifact_location=hit.artifact_location,
-                          lifecycle_stage=hit.lifecycle_stage)
+        return Experiment(experiment_id=hit["_id"], name=hit["_source"]["name"],
+                          artifact_location=hit["_source"]["artifact_location"],
```
Comment (jcuquemelle) [inline mlflow_elasticsearchstore/elasticsearch_store.py:65]: wondering if it is normal for client code to access an underscored field, seems weird according to python coding rules

File: mlflow_elasticsearchstore/models.py [hunk 2]
```diff
@@ -128,3 +129,61 @@ def to_mlflow_entity(self) -> Run:
...
+        if not (indices.exists(index=[cls.name])):
+            indices.create(index=cls.name,
+                           body={"settings": cls.settings,
+                                 "mappings": cls.mappings})
+
+
+class RunIndex():
+    name = "mlflow-runs"
+    settings = {
+        "number_of_shards": 2,
```
Comment (jcuquemelle) [inline mlflow_elasticsearchstore/models.py:160]: Why do we chose these values for shards and replicas ? (and why 1 only in the ExperimentIndex ?)

Comment (AxelVivien25) [reply mlflow_elasticsearchstore/models.py:160]: Run Index is bigger than Experiment Index but values are a bit random

File: mlflow_elasticsearchstore/elasticsearch_store.py (hunk 1 above)
Comment (AxelVivien25) [reply mlflow_elasticsearchstore/elasticsearch_store.py:65]: The object elasticsearch returns is made that way. If I want to access my object I have to look into _source.

File: mlflow_elasticsearchstore/elasticsearch_store.py [hunk 3]
```diff
@@ -50,69 +53,88 @@ class ElasticsearchStore(AbstractStore):
...
-                       status=hit.status,
-                       start_time=hit.start_time,
-                       end_time=hit.end_time if hasattr(hit, 'end_time') else None,
-                       lifecycle_stage=hit.lifecycle_stage, artifact_uri=hit.artifact_uri)
+        return RunInfo(run_uuid=hit["_id"], run_id=hit["_id"],
+                       experiment_id=hit["_source"]["experiment_id"],
+                       user_id=hit["_source"]["user_id"],
+                       status=hit["_source"]["status"],
+                       start_time=hit["_source"]["start_time"],
+                       end_time=hit["_source"]["end_time"] if hasattr(hit, 'end_time') else None,
```
[the remark this answers is not in the document]
Comment (AxelVivien25) [reply mlflow_elasticsearchstore/elasticsearch_store.py:78]: I change this on the commit where I implement run features. You want me to change it here ?

File: Dockerfile [hunk 4]
```diff
@@ -1,4 +1,4 @@
-FROM elasticsearch:6.8.11
+FROM elasticsearch:7.9.0
```
Comment (t-henri) [inline Dockerfile:1]: I think the version we have in our cluster is 7.7, could you take an image with the same version please

File: mlflow_elasticsearchstore/models.py (hunk 2 above)
Comment (t-henri) [reply mlflow_elasticsearchstore/models.py:160]: This explanation fits for the number of shards, but why do you replicate the runs and not the experiments ?

File: tests/integration_tests/test_integration_elasticsearchstore.py [hunk 5]
```diff
@@ -121,6 +123,7 @@ def test_set_experiment_tag_of_non_active_experiment(init_store):
         assert "must be in the 'active' state" in str(excinfo.value)
```
Comment (t-henri) [inline tests/integration_tests/test_integration_elasticsearchstore.py:123]: Isn't this test and experiment feature ?

File: tests/unit_tests/test_elasticsearchstore.py [hunk 6]
```diff
@@ -57,64 +70,81 @@
...
+        "query": {
+            "bool": {
+                "filter": [
+                    {"terms": {"lifecycle_stage": ['active']}}
+                ]
+            }
+        }
+    }
+    search_mock.assert_called_once_with(index=ExperimentIndex.name, body=body)
+    mock_experiments = [create_store._hit_to_mlflow_experiment(e) for e in response["hits"]["hits"]]
```
Comment (t-henri) [inline tests/unit_tests/test_elasticsearchstore.py:94]: this is not a mock, please rename the variable

File: tests/integration_tests/test_integration_elasticsearchstore.py (hunk 5 above)
Comment (AxelVivien25) [reply tests/integration_tests/test_integration_elasticsearchstore.py:123]: set_experiment_tag function is not implemented yet

File: Dockerfile (hunk 4 above)
Comment (AxelVivien25) [reply Dockerfile:1]: I did this on the last commit. DB dump is version specific so I had to redo it and I had only the code for the last mapping

**Pull request closed without merging**